integration/spark-common-test/src/test/scala/org/apache/carbondata/spark/testsuite/localdictionary/LocalDictionarySupportCreateTableTest.scala

```diff
@@ -2585,67 +2585,445 @@ class LocalDictionarySupportCreateTableTest extends QueryTest with BeforeAndAfte
+    }
+    intercept[Exception] {
+      sql("alter table local1 set tblproperties('local_dictionary_include'='add')")
+    }
+  }
+
+  test("test alter set for local dictionary exclude valid and invalid")
+  {
+    sql("drop table if exists local1")
+    sql(
+      """
+        | CREATE TABLE local1(id int, name string, city string, age int,add string)
+        | STORED BY 'carbondata' tblproperties('local_dictionary_exclude'='city','dictionary_include'='add')
+      """.stripMargin)
+
+    val descLoc1 = sql("describe formatted local1").collect
+    descLoc1.find(_.get(0).toString.contains("Local Dictionary Exclude")) match {
+      case Some(row) => assert(row.get(1).toString.contains("city"))
+    }
+    sql("alter table local1 set tblproperties('local_dictionary_exclude'='name')")
+    val descLoc2 = sql("describe formatted local1").collect
+    descLoc2.find(_.get(0).toString.contains("Local Dictionary Exclude")) match {
+      case Some(row) => assert(row.get(1).toString.contains("name"))
+    }
+    intercept[Exception] {
+      sql("alter table local1 set tblproperties('local_dictionary_exclude'='id')")
+    }
+    intercept[Exception] {
+      sql("alter table local1 set tblproperties('local_dictionary_exclude'='city,city')")
+    }
+    intercept[Exception] {
+      sql("alter table local1 set tblproperties('local_dictionary_exclude'='')")
+    }
+    intercept[Exception] {
+      sql("alter table local1 set tblproperties('local_dictionary_exclude'='add')")
+    }
+  }
+
+  test("test alter set same column for local dictionary exclude and include")
   {
     sql("drop table if exists local1")
     sql(
       """
         | CREATE TABLE local1(id int, name string, city string, age int)
-        | STORED BY 'carbondata'
+        | STORED BY 'carbondata' tblproperties('local_dictionary_exclude'='city')
       """.stripMargin)
+    intercept[Exception] {
+      sql(
+        "alter table local1 set tblproperties('local_dictionary_include'='name'," +
+        "'local_dictionary_exclude'='name')")
+    }
+  }
 
+  test("test alter set for valid and invalid complex type as include/exclude") {
+    sql("drop table if exists local1")
+    sql(
+      """
+        | CREATE TABLE local1(id int, name string, city string, age int,st struct<s_id:int,
+        | s_name:string,s_city:array<string>>, dcity array<string>)
+        | STORED BY 'org.apache.carbondata.format'
+        | tblproperties('local_dictionary_exclude'='name','local_dictionary_include'='city,dcity',
+        | 'local_dictionary_enable'='true')
+      """.
+        stripMargin)
+    val descFormatted1 = sql("describe formatted local1").collect
+    descFormatted1.find(_.get(0).toString.contains("Local Dictionary Enabled")) match {
+      case Some(row) => assert(row.get(1).toString.contains("true"))
+    }
+    descFormatted1.find(_.get(0).toString.contains("Local Dictionary Include")) match {
+      case Some(row) => assert(row.get(1).toString.contains("city,dcity"))
```

Comment:
add check to verify that name is not included in local dictionary include property.